examples/tls-with-aws-pca-issuer/main.tf

```diff
@@ -0,0 +1,233 @@
+
+  vpc_cidr     = "10.0.0.0/16"
+  vpc_name     = join("-", [local.tenant, local.environment, local.zone, "vpc"])
+  cluster_name = join("-", [local.tenant, local.environment, local.zone, "eks"])
+
+  terraform_version = "Terraform v1.0.1"
+
+  certificate_name = "example"
+  certificate_dns = "example.com"
+}
+
+module "aws_vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "v3.2.0"
+
+  name = local.vpc_name
+  cidr = local.vpc_cidr
+  azs  = data.aws_availability_zones.available.names
+
+  public_subnets  = [for k, v in data.aws_availability_zones.available.names : cidrsubnet(local.vpc_cidr, 8, k)]
+  private_subnets = [for k, v in data.aws_availability_zones.available.names : cidrsubnet(local.vpc_cidr, 8, k + 10)]
+
+  enable_nat_gateway   = true
+  create_igw           = true
+  enable_dns_hostnames = true
+  single_nat_gateway   = true
+
+  public_subnet_tags = {
+    "kubernetes.io/cluster/${local.cluster_name}" = "shared"
+    "kubernetes.io/role/elb"                      = "1"
+  }
+
+  private_subnet_tags = {
+    "kubernetes.io/cluster/${local.cluster_name}" = "shared"
+    "kubernetes.io/role/internal-elb"             = "1"
+  }
+}
+#---------------------------------------------------------------
+# Example to consume aws-eks-accelerator-for-terraform module
+#---------------------------------------------------------------
+module "aws-eks-accelerator-for-terraform" {
+  source = "../.."
+
+  tenant            = local.tenant
+  environment       = local.environment
+  zone              = local.zone
+  terraform_version = local.terraform_version
+
+  # EKS Cluster VPC and Subnet mandatory config
+  vpc_id             = module.aws_vpc.vpc_id
+  private_subnet_ids = module.aws_vpc.private_subnets
+
+  # EKS CONTROL PLANE VARIABLES
+  kubernetes_version = local.kubernetes_version
+
+  # EKS MANAGED NODE GROUPS
+  managed_node_groups = {
+    mg_4 = {
+      node_group_name = "managed-ondemand"
+      instance_types  = ["m4.large"]
+      min_size        = "2"
+      subnet_ids      = module.aws_vpc.private_subnets
+    }
+  }
+
+  depends_on = [module.aws_vpc]
+}
+
+module "kubernetes-addons" {
+  source         = "../../modules/kubernetes-addons"
+  eks_cluster_id = module.aws-eks-accelerator-for-terraform.eks_cluster_id
+  aws_privateca_acmca_arn = aws_acmpca_certificate_authority.example.arn
+
+  # EKS Managed Add-ons
+  enable_amazon_eks_vpc_cni    = true
+  enable_amazon_eks_coredns    = true
+  enable_amazon_eks_kube_proxy = true
+
+  #K8s Add-ons
+  enable_cert_manager                 = true
+  enable_aws_privateca_issuer         = true
+
+  depends_on = [module.aws-eks-accelerator-for-terraform.managed_node_groups]
+}
+
+resource "aws_acmpca_certificate_authority_certificate" "example" {
+  certificate_authority_arn = aws_acmpca_certificate_authority.example.arn
+
+  certificate       = aws_acmpca_certificate.example.certificate
+  certificate_chain = aws_acmpca_certificate.example.certificate_chain
+}
+
```

Comment:
It would be nice to add what this resource does. e.g., 
```
#-------------------------------
#  This resource issues a certificate using AWS Certificate Manager Private Certificate Authority (ACM PCA)
#-------------------------------
```